packages/salesforce-adapter/src/custom_references/profiles.ts

```diff
@@ -284,7 +308,58 @@ const getProfilesCustomReferences: WeakReferencesHandler['findWeakReferences'] =
     return refs
   }
 
+const profileEntriesTargets = (profile: InstanceElement): Dictionary<ElemID> =>
+  _(
+    mapProfileSections(
+      profile,
+      (sectionName, sectionEntryKey, target, sourceField): [string, ElemID] => [
+        [sectionName, sectionEntryKey, ...makeArray(sourceField)].join('.'),
+        target,
+      ],
+    ),
+  )
+    .fromPairs()
+    .value()
+
+const removeMissingReferences: WeakReferencesHandler['removeWeakReferences'] =
+  ({ elementsSource }) =>
+  async (elements) => {
+    const profiles = elements.filter(
+      isInstanceOfTypeSync(PROFILE_METADATA_TYPE),
+    )
+    const entriesTargets: Dictionary<ElemID> = _.merge(
+      {},
+      ...profiles.map(profileEntriesTargets),
+    )
+    const brokenReferenceFields = Object.keys(
+      await pickAsync(
+        entriesTargets,
+        async (target) => !(await elementsSource.has(target)),
+      ),
+    )
+    const brokenElements = profiles.filter((profile) =>
+      brokenReferenceFields.some((field) => _(profile.value).has(field)),
+    )
+    const fixedElements = brokenElements.map((profile) => {
+      const fixed = profile.clone()
+      fixed.value = _.omit(fixed.value, ...brokenReferenceFields)
```

Comment:
```suggestion
      fixed.value = _.omit(fixed.value, brokenReferenceFields)
```
It's better to avoid spread operator on possibly large arrays, as spread happens in the stack and can lead to overflow. (OOM exception)

Reply:
Done. Didn't realise I can just pass the array as-is.